sig-architecture/charter.md

```diff
@@ -1,40 +1,56 @@
-ensure architectural consistency over time.
-
-The scope covers issues that span all the system's components, how
-they fit together, how they interact, etc.
-
-Specific areas of focus include:
-
-* Defining the scope of the Kubernetes project
-  * [What is (and is not) Kubernetes](https://kubernetes.io/docs/concepts/overview/what-is-kubernetes/)
-* Maintaining, evolving, and enforcing the deprecation policy
-  * [Deprecation policy](https://kubernetes.io/docs/reference/deprecation-policy/)
-* Documenting and evolving the system architecture
-  * [Kubernetes Design and Architecture](../contributors/design-proposals/architecture/architecture.md)
-* Defining and driving necessary extensibility points
-* Establishing and documenting design principles
-  * [Design principles](../contributors/design-proposals/architecture/principles.md)
-* Establishing and documenting conventions for system and user-facing APIs
-  * [API conventions](/contributors/devel/api-conventions.md)
-* Developing necessary technical review processes, such as the proposal and API review processes
-* Driving improvement of overall code organization, including github orgs and repositories
-* Educating approvers/owners of other SIGs (e.g., by holding office hours)
-
-Out of scope:
-* Issues specific to a particular component or functional area, which would be the purview
-  of some other SIG, except where they deviate from project-wide principles and conventions.
-* [Release support policy](/contributors/design-proposals/release/versioning.md)
-  is owned by SIG Release
-
-TODO:
-* Formalize decision processes
-* Document initial reviewers and approvers
-* Clarify criteria for areas out of scope for the SIG
-* Document who owns client library, build, and release artifacts
-* Document who owns conformance definition, profiles, etc.
+This charter adheres to the conventions described in the [Kubernetes Charter README] and uses
+the Roles and Organization Management outlined in [sig-governance].
+
+## Scope
+
+The Architecture SIG maintains and evolves the design principles of Kubernetes, and provides a consistent body of expertise necessary to ensure architectural consistency over time.
+
+### In scope
+
+#### Code, Binaries, Docs, and Services
+
+- *Conformance test definitions*
+- *API definitions*
+- *Architectural renderings*
+- *API conventions*
+- *Design principles*
+- *Deprecation policy*
+
+#### Cross-cutting and Externally Facing Processes
+
+- API review process
+- Conformance test review and management
+- The Kubernetes Enhancement Proposal (KEP) process and KEP reviews
+- Design documentation management
+- Deprecation policy management
+- Architectural initiative backlog management
+
+### Out of scope
+
+- KEPs that do not have architectural implications
+- The release enhancement delivery process
+
+## Roles and Organization Management
+
+This sig follows and adheres to the Roles and Organization Management outlined in [sig-governance]
```

Comment:
I'm concerned that in the past it has been fairly strongly argued, and in some cases agreed, that these are probably not appropriate for sig-arch.

For example:

https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/kubernetes-sig-architecture/FpcrJQW1R1A/Oh7dN_JEAwAJ

... and 

https://docs.google.com/document/d/1p4quG5qzIllQ52eKIb7mGrgaiRgQeiu70iJi9MZ9C8c/edit?disco=AAAAB7Yujow

... and 

https://github.com/kubernetes/community/pull/2074#discussion_r184764262

... and 

https://github.com/kubernetes/community/pull/2074#discussion_r184466503

and ...

https://github.com/kubernetes/community/pull/2074#discussion_r184762567

and...

https://github.com/kubernetes/community/pull/2074#discussion_r184762567

and...

https://docs.google.com/document/d/1p4quG5qzIllQ52eKIb7mGrgaiRgQeiu70iJi9MZ9C8c/edit?disco=AAAAB7Yuin0

This, and the fact that these concerns repeatedly get raised, and then the issues or docs where they are raised get closed and replaced by docs or PR's where these issues are not mentioned or addressed is creating the impression that we're trying to ignore or squash these concerns, and simply continue with the status quo.

I think that we should either explicitly mention these concerns and discussions as a TODO in this PR before merging it, or not merge it until these concerns and discussion have properly been resolved.

Reply:
As discussed and agreed upon in today's SIG-Architecture meeting, we should add the following wording to this PR and then merge it:

TODO: 

1. There remains lack of consensus in this SIG regarding the need for company and other diversity in the leadership of this SIG and it's subprojects, beyond what is contained in [sig-governance].  This still needs to be resolved and documented in this charter.  Further details may be found [here](https://docs.google.com/document/d/1TTcfvf8T_tBhGDm-wjgg31WrWjYg8IZEmo3b1mpUXh0/edit?usp=sharing)

2. It remains unspecified whether and how SIG-Architecture gets involved in resolving technical conflicts within and between other SIG's, and this SIG's sub-projects.  This still needs to be resolved and documented in this charter.

Reply:
Todos were added below